### DISCUSSION TITLE
[GUIDE] Step by step guide to installing FASTSDCPU on ANDROID from scratch

### DISCUSSION THREAD

**Discussion by patientx** [rupeshs/fastsdcpu#123] [Show and tell]
UPDATE : Latest update requires "mediapipe v0.10.11" which isn't yet available in android so far, the latest version is 0.10.9. Tried with it and it is working at least with default lcm model and newest sdxs model. 

1- ) Install termux from f-droid.  (https://f-droid.org/packages/com.termux/)
----  Open termux  & enter and execute these commands one by one
2- )  pkg update 
3- ) pkg upgrade -y
4- ) termux-setup-storage
5- ) pkg install wget  -y
6- ) pkg install git -y
7- ) pkg install proot -y
8- ) cd ~
9- ) git clone https://github.com/MFDGaming/ubuntu-in-termux.git
10- ) cd ubuntu-in-termux
11- ) chmod +x ubuntu.sh
12- ) ./ubuntu.sh -y
13- ) ./startubuntu.sh
---- We are in ubuntu now, you'll see that the prompt would change to root@localhost
14- ) apt update -y
15- ) apt dist-upgrade -y
16- ) apt install wget git -y
17- ) apt install python3 -y
18- ) apt install python3.10-venv -y
19- ) apt install gcc python3-dev -y
20- ) git clone https://github.com/rupeshs/fastsdcpu.git
21- ) cd fastsdcpu
22- ) chmod +x install.sh
23- ) ./install.sh --disable-gui                                
---- Now the app will install its required files, gonna take a while. After it is finished, use this to start the app in webui mode.

Added ::: 

24- ) apt-get install ffmpeg

25- ) ./start-webui.sh
---- Again, when everything is good you'll see "Running on local URL. http://127.0.0.1:7860" Then ....
26- ) Open your preferred web browser on your phone and enter the address http://127.0.0.1:7860
---- First time generating it is going to download the default model which takes around 4 GB of space so be ready spacewise and timewise ...

Annd... it is done.

On my device : (Xiaomi Red Mi Note 8 Pro) with default sd-turbo model, it takes around 45seconds for one step, also with default VAE my phone can't decode it probably not enough RAM, so I switched on "use tiny encoder on SD" in generation settings tab.

One step total time 75 seconds 
Can do two images at a total time of 105 seconds. /tried 4 nope it crashed :) so two it is.
TO RUN IT AT LATER TIMES : 

on termux ::

cd ubuntu-in-termux
./startubuntu.sh

then on ubuntu ::

cd fastsdcpu
 ./start-webui.sh

**Comment by rupeshs** [comment]
@patientx Nice,could you please fix typo in the title "ANDROID from stratch".

**Comment by patientx** [comment]
[
![image (6)](https://github.com/rupeshs/fastsdcpu/assets/5392772/cb0d867d-a4be-424c-9cf6-14b1ee97e503)
![image (14)](https://github.com/rupeshs/fastsdcpu/assets/5392772/677e781d-d1cc-4962-a7a0-4531910e6c0b)
![image (11)](https://github.com/rupeshs/fastsdcpu/assets/5392772/e36485f1-3151-42fe-bc69-4190ac18329a)
![image (9)](https://github.com/rupeshs/fastsdcpu/assets/5392772/de44e74b-4f70-40d4-95f6-a77ef330bbe4)
]

**Comment by patientx** [comment]
I wanted to tell anyone who was going to try this that there is also a way to faceswap on android natively with the same way as this. It is working very well and faster and less painless compared to fastsdcpu (requiring less resources ofc) 

https://github.com/facefusion/facefusion

You can see my guide for that on their discord, same stuff with fastsdcpu really, just a few extra steps.

QUICK TIP : On android the termux and linux stuff takes a lot of cpu power & memory making running the browser a bit slower sometimes auto closing the linux stuff because of memory errors. I found out that using a barebones browser specifically for these things (both fastsdcpu and facefusion) makes it a lot responsive and lowers the chance out of memory errors occuring. Tried a few of them on play store and finally settled on using this : https://play.google.com/store/apps/details?id=net.onecook.browser&hl=en&gl=US Of course you can use others , just use this enter the ip of facefusion save it as default page and it is very fast to open and use when you open the app compared to chrome or other big time browsers.

**Comment by QkiZMR** [comment]
```root@localhost:~/AI/fastsdcpu# bash ./install.sh --disable-gui
Starting FastSD CPU env installation...
Found python command
Python version : 3.11.8
Looking in indexes: https://download.pytorch.org/whl/cpu
ERROR: Could not find a version that satisfies the requirement torch==2.0.1 (from versions: none)
ERROR: No matching distribution found for torch==2.0.1```
There's no such version for arm64.

**Reply by ToonTamilIndia** [reply]
> ```
> Starting FastSD CPU env installation...
> Found python command
> Python version : 3.11.8
> Looking in indexes: https://download.pytorch.org/whl/cpu
> ERROR: Could not find a version that satisfies the requirement torch==2.0.1 (from versions: none)
> ERROR: No matching distribution found for torch==2.0.1```
> There's no such version for arm64.
> ```

Try to install torch manually

**Comment by patientx** [comment]
you need python3.10 I suspect. You seem to have 3.11. It doesn't work even on PC with 3.11 for me.

**Reply by Recon-77** [reply]
I think we need to build pytorch for arm64 Architecture, I also have the same Problem even in Python 3.10 on  Different Android Versions ( Android 8, Android 10, and in Android 12 )

**Comment by ToonTamilIndia** [comment]
![Screenshot_2024-06-27-20-46-46-731_com termux](https://github.com/rupeshs/fastsdcpu/assets/121746774/14993fe4-2e99-47f2-9433-696bd4d802ff)

**Comment by hubono** [comment]
![Screenshot_20240911-095355](https://github.com/user-attachments/assets/af275ede-72ab-4a41-8e6f-67afd6bd3803)

**Comment by patientx** [comment]
apt install ffmpeg

**Comment by Waiiiiiiiiiii** [comment]
![Screenshot_2025-03-27-20-44-58-51_84d3000e3f4017145260f7618db1d683](https://github.com/user-attachments/assets/8f2ac3ec-fff6-478d-a0bc-2e468c4a5c15)

**Reply by rupeshs** [reply]
> ![Screenshot_2025-03-27-20-44-58-51_84d3000e3f4017145260f7618db1d683](https://private-user-images.githubusercontent.com/96092375/427797554-8f2ac3ec-fff6-478d-a0bc-2e468c4a5c15.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMxMjQxMzgsIm5iZiI6MTc0MzEyMzgzOCwicGF0aCI6Ii85NjA5MjM3NS80Mjc3OTc1NTQtOGYyYWMzZWMtZmZmNi00NzhkLWEwYmMtMmU0NjhjNGE1YzE1LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI4VDAxMDM1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTczNmVlY2I4NTg2OTM2YjBhODk5ZmQ4NGYyMzRhODQ0M2ZmNjFhOWI2MzJjNTI1YjMwZDZmODdkZGJlMGY1YmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-_l5On1nMVG3jsuKai9JXA8izpD3MPKxmIiRmXgb4-k)

Try this https://nolowiz.com/how-to-install-and-run-fastsd-cpu-on-android-temux-step-by-step-guide/

**Reply by Waiiiiiiiiiii** [reply]
I tried that method before. It dies here.
![Screenshot_2025-03-27-21-29-42-41_84d3000e3f4017145260f7618db1d683](https://github.com/user-attachments/assets/dfcd5c5b-97a8-4970-842f-33e91542f573)

**Reply by Nick088Official** [reply]
> I tried that method before. It dies here. ![Screenshot_2025-03-27-21-29-42-41_84d3000e3f4017145260f7618db1d683](https://private-user-images.githubusercontent.com/96092375/427816715-dfcd5c5b-97a8-4970-842f-33e91542f573.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMxNjc0NDYsIm5iZiI6MTc0MzE2NzE0NiwicGF0aCI6Ii85NjA5MjM3NS80Mjc4MTY3MTUtZGZjZDVjNWItOTdhOC00OTcwLTg0MmYtMzNlOTE1NDJmNTczLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI4VDEzMDU0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZiOTEwNDljZWEwOTM3NzBjZjJmNzg1YzM0ZDg0MDJjMmU2ZDBlZjAwZTZjZDJkMzBmNzU0ODYwYjc4OGVkZDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WVd6Z-NC2eL3BTh4qtzAuipxWO55vIZB4ByTBgrfnj4)

The guide Is outdated, use https://github.com/rupeshs/fastsdcpu/discussions/278

**Comment by rupeshs** [comment]
Starting from v1.0.0-beta.200 we will use uv for installation 
Updated installation guide : https://nolowiz.com/how-to-install-and-run-fastsd-cpu-on-android-temux-step-by-step-guide/
Release : https://github.com/rupeshs/fastsdcpu/releases/tag/v1.0.0-beta.200
